File: DynamicKey/AgoraDynamicKey/ruby/sample/rtm_token_builder_sample.rb
require_relative '../lib/dynamic_key'

# Need to set environment variable AGORA_APP_ID
app_id = ENV['AGORA_APP_ID']
# Need to set environment variable AGORA_APP_CERTIFICATE
app_certificate = ENV['AGORA_APP_CERTIFICATE']

expiration_time_in_seconds = 3600
current_time_stamps = Time.now.to_i

puts "App Id: #{app_id}"
puts "App Certificate: #{app_certificate}"
if !app_id || app_id == '' || !app_certificate || app_certificate == ''
  puts 'Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE'
  exit
end

params = {
  app_id: app_id,
  app_certificate: app_certificate,
  account: 'test_user_id',
  role: AgoraDynamicKey::RTMTokenBuilder::Role::RTM_USER,
  privilege_expired_ts: current_time_stamps + expiration_time_in_seconds
}

result = AgoraDynamicKey::RTMTokenBuilder.build_token params
puts "Rtm Token: #{result}"

File: DynamicKey/AgoraDynamicKey/nodejs/sample/RtmTokenBuilderSample.js
const RtmTokenBuilder = require("../src/RtmTokenBuilder").RtmTokenBuilder;
const RtmRole = require("../src/RtmTokenBuilder").Role;
const Priviledges = require("../src/AccessToken").priviledges;

// Need to set environment variable AGORA_APP_ID
const appId = process.env.AGORA_APP_ID;
// Need to set environment variable AGORA_APP_CERTIFICATE
const appCertificate = process.env.AGORA_APP_CERTIFICATE;

const account = "test_user_id";
const expirationTimeInSeconds = 3600;
const currentTimestamp = Math.floor(Date.now() / 1000);
const privilegeExpiredTs = currentTimestamp + expirationTimeInSeconds;

console.log("App Id:", appId);
console.log("App Certificate:", appCertificate);
if (appId == undefined || appId == "" || appCertificate == undefined || appCertificate == "") {
    console.log("Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE");
    process.exit(1);
}

const token = RtmTokenBuilder.buildToken(appId, appCertificate, account, RtmRole, privilegeExpiredTs);
console.log("Rtm Token:", token);
